### DISCUSSION TITLE
PDF not including external stylesheets or images.

### DISCUSSION THREAD

**Discussion by thursdaydan** [spatie/browsershot#662] [Q&A] [answered]
Hi all, 

Got a bit of a strange one. I have a queue job that generates a PDF but external stylesheets and images are not being included. 

However, when running my queue locally via artisan it works. Also generating the PDF via a route in the browser works. Wondered if anyone else has come across this issue before.

I hope it's me being a bit silly!

Here is an example of how I'm calling Browsershot.

```
$content = view('site-report.drawing-template', ['drawing' => $drawing])->render();
Browsershot::html($content)
  ->margins(0, 2, 0, 2)
  ->format('A4')
  ->noSandbox()
  ->setOption('args', ['--disable-web-security'])
  ->showBackground()
  ->waitUntilNetworkIdle()
  ->save($targetFullPath);
```

Finally here is the header for the PDF template blade file I'm using.
```
    <head>
        <meta charset="utf-8">
        <meta name="viewport" content="width=device-width, initial-scale=1">
        <meta name="csrf-token" content="{{ csrf_token() }}">

        <title>{{ config('app.name', 'Laravel') }}</title>

        <!-- Fonts -->
        <link rel="stylesheet" href="https://fonts.googleapis.com/css2?family=Nunito:wght@400;600;700&display=swap">

        <!-- Styles -->
        <link rel="stylesheet" href="{{ asset('css/pdf.css') }}">

        <!-- Scripts -->
        <script src="{{ asset('js/app.js') }}"></script>
    </head>
```

**Comment by iBet7o** [comment]
Hi @thursdaydan

did you find the solution to this problem?

**Reply by abbasmashaddy72** [reply]
You have to use public_path, but not asset for CSS and JS to work.

**Comment by abbasmashaddy72** [comment, marked as answer]
You have to use public_path, but not asset for CSS and JS to work.

**Reply by thursdaydan** [reply]
Thanks, tried with public path and it worked fine! Was it something I missed from the docs in laravel do you know, or is it just from trial and error/experience?

**Comment by hyphen81** [comment]
You mentioned in the title that it wasn't rendering images, which is what I'm having an issue with. Even trying public_path, the image is broken. Anyone know how to get that working?

**Comment by RyanGSampson** [comment]
My issue was that my dev server has a username and password security rule in place to access the site.
If anyone is developing on forge and has this issue, turn off your site security rule and see if that fixes it for you too.
